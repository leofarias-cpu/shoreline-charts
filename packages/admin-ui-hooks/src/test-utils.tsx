import 'jest-location-mock'

export function jestWindowMock() {
  let history: string[] = [window.location.pathname]
  let index = 0
  let popstateCallback = () => {}

  Object.defineProperties(window.history, {
    pushState: {
      value: (data: { path: string }) => {
        history = [...history.splice(0, index + 1), data.path]
        index += 1

        window.location.assign(data.path)
        popstateCallback()
      },
    },
    back: {
      value: () => {
        if (index <= 0) return

        index -= 1
        window.location.assign(history[index])
        popstateCallback()
      },
    },
    forward: {
      value: () => {
        if (index >= history.length) return

        index += 1
        window.location.assign(history[index])
        popstateCallback()
      },
    },
  })
  Object.defineProperties(window, {
    addEventListener: {
      value: (name: string, fn: () => void) => {
        if (name === 'popstate') {
          popstateCallback = fn
        }
      },
    },
  })
}

export function setQueryParam(query: Record<string, string | undefined>) {
  const queryParams = new URLSearchParams(window.location.search)

  Object.entries(query).forEach((element: [string, string | undefined]) => {
    if (!element[1]) {
      queryParams.delete(element[0])

      return
    }

    queryParams.set(element[0], element[1])
  })

  window.location.assign(`?${queryParams.toString()}`)
}

export function getCurrentQuery() {
  const queryParams = new URLSearchParams(window.location.search)

  return Object.fromEntries(queryParams.entries())
}

export function cleanQueryParam() {
  window.location.assign('')
}
