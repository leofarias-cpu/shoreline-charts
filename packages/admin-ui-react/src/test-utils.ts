import type { ReactElement } from 'react'
import type { RenderOptions } from '@testing-library/react'
import { render as baseRender } from '@testing-library/react'
import { ThemeProvider } from './system'

function render(ui: ReactElement, options?: Omit<RenderOptions, 'queries'>) {
  return baseRender(ui, { wrapper: ThemeProvider, ...options })
}

export { render }
