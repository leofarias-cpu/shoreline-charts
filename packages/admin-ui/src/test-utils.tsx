import React, { useRef } from 'react'
import { axe } from 'jest-axe'
import type { RenderOptions } from '@testing-library/react'
import { render as baseRender } from '@testing-library/react'
import { ThemeProvider } from '@vtex/admin-core'
import type { ReactElement } from 'react'
import { renderHook, act } from '@testing-library/react-hooks'

function render(ui: ReactElement, options?: Omit<RenderOptions, 'queries'>) {
  return baseRender(ui, { wrapper: ThemeProvider, ...options })
}

/**
 * HOC to wrap a component with its state hook
 * @example
 * // Cmp has the useCmp state hook
 * const StatefulCpm = withState(Cmp, () => useCmpState({ isSelected: false }))
 */
function withState(Component: any, useStateHook: () => any) {
  function StatefulComponent(props: any) {
    // Avoids an invalid hook call
    const ref = useRef(useStateHook())

    return <Component state={ref.current} {...props} />
  }

  StatefulComponent.displayName = Component.displayName

  return StatefulComponent
}

function jestMatchMedia() {
  /**
   * ðŸš§ Workaround for window.match media
   * @see https://jestjs.io/docs/en/manual-mocks#mocking-methods-which-are-not-implemented-in-jsdom
   */
  return Object.defineProperty(window, 'matchMedia', {
    writable: true,
    value: jest.fn().mockImplementation((query) => ({
      matches: false,
      media: query,
      onchange: null,
      addListener: jest.fn(), // Deprecated
      removeListener: jest.fn(), // Deprecated
      addEventListener: jest.fn(),
      removeEventListener: jest.fn(),
      dispatchEvent: jest.fn(),
    })),
  })
}

export * from '@testing-library/react'
export { render, withState, axe, jestMatchMedia, renderHook, act }
