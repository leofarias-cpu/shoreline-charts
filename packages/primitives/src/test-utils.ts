import type { RenderOptions } from '@testing-library/react'
import { render as baseRender } from '@testing-library/react'
import { ThemeProvider } from '@vtex/admin-core'
import type { ReactElement } from 'react'

function render(ui: ReactElement, options?: Omit<RenderOptions, 'queries'>) {
  return baseRender(ui, { wrapper: ThemeProvider, ...options })
}

export * from '@testing-library/react'
export { render }
