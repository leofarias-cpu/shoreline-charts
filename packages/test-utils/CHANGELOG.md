# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [1.0.0-rc.13](https://github.com/vtex/shoreline/compare/@vtex/shoreline-test-utils@1.0.0-rc.12...@vtex/shoreline-test-utils@1.0.0-rc.13) (2024-03-01)

**Note:** Version bump only for package @vtex/shoreline-test-utils

# [1.0.0-rc.12](https://github.com/vtex/shoreline/compare/@vtex/shoreline-test-utils@1.0.0-rc.11...@vtex/shoreline-test-utils@1.0.0-rc.12) (2024-03-01)

**Note:** Version bump only for package @vtex/shoreline-test-utils

# [1.0.0-rc.11](https://github.com/vtex/shoreline/compare/@vtex/shoreline-test-utils@1.0.0-rc.10...@vtex/shoreline-test-utils@1.0.0-rc.11) (2024-02-29)

**Note:** Version bump only for package @vtex/shoreline-test-utils

# [1.0.0-rc.10](https://github.com/vtex/shoreline/compare/@vtex/shoreline-test-utils@1.0.0-rc.9...@vtex/shoreline-test-utils@1.0.0-rc.10) (2024-02-29)

**Note:** Version bump only for package @vtex/shoreline-test-utils

# [1.0.0-rc.9](https://github.com/vtex/shoreline/compare/@vtex/shoreline-test-utils@1.0.0-rc.8...@vtex/shoreline-test-utils@1.0.0-rc.9) (2024-02-23)

**Note:** Version bump only for package @vtex/shoreline-test-utils

# [1.0.0-rc.8](https://github.com/vtex/shoreline/compare/@vtex/shoreline-test-utils@1.0.0-rc.7...@vtex/shoreline-test-utils@1.0.0-rc.8) (2024-02-23)

**Note:** Version bump only for package @vtex/shoreline-test-utils

# [1.0.0-rc.7](https://github.com/vtex/shoreline/compare/@vtex/shoreline-test-utils@1.0.0-rc.6...@vtex/shoreline-test-utils@1.0.0-rc.7) (2024-02-22)

**Note:** Version bump only for package @vtex/shoreline-test-utils

# [1.0.0-rc.6](https://github.com/vtex/shoreline/compare/@vtex/shoreline-test-utils@1.0.0-rc.5...@vtex/shoreline-test-utils@1.0.0-rc.6) (2024-02-09)

**Note:** Version bump only for package @vtex/shoreline-test-utils

# [1.0.0-rc.5](https://github.com/vtex/shoreline/compare/@vtex/shoreline-test-utils@1.0.0-rc.4...@vtex/shoreline-test-utils@1.0.0-rc.5) (2024-02-09)

**Note:** Version bump only for package @vtex/shoreline-test-utils

# [1.0.0-rc.4](https://github.com/vtex/shoreline/compare/@vtex/shoreline-test-utils@1.0.0-rc.3...@vtex/shoreline-test-utils@1.0.0-rc.4) (2024-02-09)

**Note:** Version bump only for package @vtex/shoreline-test-utils

# [1.0.0-rc.3](https://github.com/vtex/shoreline/compare/@vtex/shoreline-test-utils@1.0.0-rc.2...@vtex/shoreline-test-utils@1.0.0-rc.3) (2024-02-08)

**Note:** Version bump only for package @vtex/shoreline-test-utils

# [1.0.0-rc.2](https://github.com/vtex/shoreline/compare/@vtex/shoreline-test-utils@1.0.0-rc.1...@vtex/shoreline-test-utils@1.0.0-rc.2) (2024-02-08)

**Note:** Version bump only for package @vtex/shoreline-test-utils

# [1.0.0-rc.1](https://github.com/vtex/shoreline/compare/@vtex/shoreline-test-utils@1.0.0-rc.0...@vtex/shoreline-test-utils@1.0.0-rc.1) (2024-02-08)

**Note:** Version bump only for package @vtex/shoreline-test-utils

# [1.0.0-rc.0](https://github.com/vtex/shoreline/compare/@vtex/shoreline-test-utils@0.3.0...@vtex/shoreline-test-utils@1.0.0-rc.0) (2024-02-08)

**Note:** Version bump only for package @vtex/shoreline-test-utils

# [0.3.0](https://github.com/vtex/shoreline/compare/@vtex/shoreline-test-utils@0.2.0...@vtex/shoreline-test-utils@0.3.0) (2023-12-05)

### Features

- **store:** add package ([fb3d946](https://github.com/vtex/shoreline/commit/fb3d946f021c914cc30d939567276300d39efdc0))

# [0.2.0](https://github.com/vtex/shoreline/compare/@vtex/shoreline-test-utils@0.1.0...@vtex/shoreline-test-utils@0.2.0) (2023-11-29)

### Features

- **tsup:** add tsup to global scope ([1efec54](https://github.com/vtex/shoreline/commit/1efec5401933eb1d06651e60ebd087a2a1c18e35))

# 0.1.0 (2023-11-29)

### Features

- **test-utils:** create package ([390fe1e](https://github.com/vtex/shoreline/commit/390fe1eb58d49281a477fafc19b5c345f20fda0b))
