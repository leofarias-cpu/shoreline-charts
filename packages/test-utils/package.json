{
  "name": "@vtex/shoreline-test-utils",
  "version": "0.0.0",
  "description": "Shoreline test utilities",
  "main": "./dist/index.mjs",
  "module": "./dist/index.mjs",
  "types": "./dist/index.d.mts",
  "repository": "vtex/shoreline",
  "publishConfig": {
    "access": "public",
    "registry": "https://registry.npmjs.org"
  },
  "files": [
    "dist"
  ],
  "engines": {
    "node": ">=12"
  },
  "scripts": {
    "prebuild": "rm -rf dist",
    "dev": "tsup --watch",
    "build": "npm run prebuild && tsup"
  },
  "peerDependencies": {
    "vitest": "0.x"
  },
  "devDependencies": {
    "tsup": "7.2.0",
    "vitest": "0.34.6"
  },
  "dependencies": {
    "@testing-library/dom": "9.3.3",
    "@testing-library/jest-dom": "6.1.4",
    "@testing-library/react": "14.1.2"
  }
}
