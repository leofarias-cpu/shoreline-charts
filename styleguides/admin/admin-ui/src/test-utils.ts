import { render as baseRender, RenderOptions } from '@testing-library/react'
import { ThemeProvider } from '@vtex/admin-core'
import { ReactElement } from 'react'

function render(ui: ReactElement, options?: Omit<RenderOptions, 'queries'>) {
  return baseRender(ui, { wrapper: ThemeProvider, ...options })
}

export * from '@testing-library/react'
export { render }
