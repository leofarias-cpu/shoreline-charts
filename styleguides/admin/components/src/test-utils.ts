import { render as baseRender, RenderOptions } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import serializer, { matchers } from 'jest-emotion'
import { axe, toHaveNoViolations } from 'jest-axe'
import { ThemeProvider } from '@vtex/admin-core'
import { ReactElement } from 'react'

expect.addSnapshotSerializer(serializer)
expect.extend(matchers)
expect.extend(toHaveNoViolations)

function render(ui: ReactElement, options?: Omit<RenderOptions, 'queries'>) {
  return baseRender(ui, { wrapper: ThemeProvider, ...options })
}

export * from '@testing-library/react'
export { render, axe }
