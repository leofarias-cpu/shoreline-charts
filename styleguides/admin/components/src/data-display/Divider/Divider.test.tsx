import React, { Fragment } from 'react'

import { Divider } from './index'
import { render, axe } from '../../test-utils'

describe('Divider tests', () => {
  it('should have overridable styles', () => {
    const { getByTestId } = render(
      <Divider data-testid="divider" csx={{ bg: 'azure' }} />
    )

    expect(getByTestId('divider')).toHaveStyleRule('background-color', 'azure')
  })

  it('should match snapshot', () => {
    const { asFragment } = render(
      <Fragment>
        <Divider />
        <Divider orientation="vertical" />
      </Fragment>
    )

    expect(asFragment()).toMatchSnapshot()
  })

  it('should not have a11y violations', async () => {
    const { container } = render(
      <Fragment>
        <Divider />
        <Divider orientation="vertical" />
      </Fragment>
    )

    const results = await axe(container)

    expect(results).toHaveNoViolations()
  })
})
