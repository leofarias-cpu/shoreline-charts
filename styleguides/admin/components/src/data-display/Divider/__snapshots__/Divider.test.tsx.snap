// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider tests should match snapshot 1`] = `
<DocumentFragment>
  .emotion-0 {
  font-family: VTEXTrustVF,-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  line-height: 1.5;
  font-variation-settings: 'wght' 92;
  font-size: 1.25rem;
  border: solid;
  border-width: 1px;
  border-color: #E0E2E7;
  margin: 0rem;
}

<hr
    aria-orientation="horizontal"
    class="emotion-0"
    role="separator"
  />
  .emotion-0 {
  font-family: VTEXTrustVF,-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  line-height: 1.5;
  font-variation-settings: 'wght' 92;
  font-size: 1.25rem;
  border: solid;
  border-width: 1px;
  border-color: #E0E2E7;
  margin: 0rem;
  border-left: 0;
  height: auto;
}

<hr
    aria-orientation="vertical"
    class="emotion-0"
    role="separator"
  />
</DocumentFragment>
`;
